### PR TITLE
Warn about empty TM master actions 

### LINKED ISSUE #14471: Throw a warning/error on empty TensorMechanics/Master action blocks with parameters
## Reason
I encountered (wrote) an input file with a block of the form

```
[Modules/TensorMechancs/Master]
  add_variables = true
  strain = SMALL
[]
```

that ended up silently doing nothing. The correct syntax would have been

```
[Modules/TensorMechancs/Master]
  [./all]
    add_variables = true
    strain = SMALL
  [../]
[]
```

The top level parameters are accepted by the action to be applied to all sub blocks of the action.

## Design
If there are no sub blocks we should throw an unused parameter warning.

## Impact
Better error reporting

Tag @sapitts

### PR DESCRIPTION
**Pull request by dschwen**
Closes #14471

### CODE REVIEW

**Comment by dschwen** [comment]
SAM build failure is unrelated.

**Review by sapitts — APPROVED** [review]
Looks good to me. Please merge @permcody -- this would have saved me at least half an hour of much frustration this week

File: modules/tensor_mechanics/test/tests/action/tests [hunk 1]
```diff
@@ -45,6 +45,15 @@
     requirement = 'The TensorMechanics MasterAction shall error if an input file specifies overlapping block restrictions for the MasterAction in input files with more than one instance of the MasterAction block.'
   [../]
 
+  [./no_block]
+    type = RunException
+    input = 'no_block.i'
+    cli_args = '--error'
+    expect_err = 'Common parameters are supplied, but not used in'
+    design = 'syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The TensorMechanics MasterAction shall warn if global Master action parameters are supplied but no Master action subblock have been added.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/action/tests]: have->has

**Review by bwspenc — APPROVED** [review]
Looks good! I won't let my minor quibble about the requirement hold this up.

**Pull request merged**